pkg/utils/atomic/slice.go

```diff
@@ -0,0 +1,64 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package atomic
```

Comment:
This looks pretty similar to https://github.com/deckarep/golang-set/blob/main/threadsafe.go. can we use this instead of implementing our own?

Reply:
I think there's a nice benefit in that ours is typed based on generic types and theirs isn't. I'd be inclined to stick to our implementation. Also, we use it in our testing.

Reply:
Yeah. That's good with me. I think we could move all of the atomic util code over to this file, but don't think it's an issue.